probe-rs/src/bin/probe-rs/util/common_options.rs

```diff
@@ -569,7 +569,26 @@ impl From<std::io::Error> for OperationError {
 
 impl From<OperationError> for DebuggerError {
     fn from(e: OperationError) -> Self {
-        DebuggerError::Other(e.into())
+        match e {
+            OperationError::AttachingFailed {
+                source,
+                connect_under_reset,
+            } => match source {
+                probe_rs::Error::Timeout => {
+                    if !connect_under_reset {
+                        DebuggerError::UserMessage(
```

Comment:
I know it's the least important point, but is there a way to write this so that rustfmt doesn't trip up on the function?

Reply:
imo its very important! makes devs mote happy :)

Reply:
Sure, but I'd still appreciate an opinion on this question (which imo is even more important) ... "This feels like a rather cumbersome way of propagating information about an error from the api to the application.
Is there a better approach than I took here?"

Reply:
Ah, yes, ofc :) I am not sure I know a better way :/ What do you dislike specially? :)

Reply:
@Yatekii Here's my logic.
- probe-rs api has (a lot of) very specific error variants.
- Authors of the api often has the best knowledge of what caused those errors, and what might fix them.
- Yet, every consumer has to recreate that knowledge ...

For the connect under reset example, cargo-flash, cargo-embed, and now the debugger, each have custom code, to trap the same error, and try to guide the user to a resolution.

 In an ideal world, we could do ...
A) ... the auto retry (that disappeared after #1259 introduced it), or if for some reason that was deemed unpalatable...
B) ... then my question would be, can we somehow capture the knowledge of cause and solution in the api, so that all consumers have access to it?